src/path_gui.cpp

```diff
@@ -228,6 +229,25 @@ PathBuildGui::~PathBuildGui()
 	this->SetSelector(nullptr);
 }
 
+/**
+ * Try to add (if LMB pressed) or remove (if RMB pressed) a path tile at the vpxel pointed to by the mouse.
+ * @param m_state Mouse button state.
+ */
+void PathBuildGui::TryAddRemovePath(uint8 m_state)
+{
+	if ((m_state & (MB_LEFT | MB_RIGHT)) == 0) return; // No buttons pressed.
+
+	const Voxel *v = _world.GetVoxel(this->mouse_pos);
+	if (!v) return; // No voxel -> no ground there.
+
+	uint16 instance = v->GetInstance();
+	if (instance == SRI_FREE && (m_state & MB_LEFT) != 0) {
+		this->BuildSinglePath(this->mouse_pos);
+	} else if (instance == SRI_PATH && (m_state & MB_RIGHT) != 0) {
```

Comment:
We will probably also want to allow for changing the current path type with [ChangePathTile](https://github.com/FreeRCT/FreeRCT/blob/master/src/path_build.cpp#L288).